pkg/bloomgateway/util.go

```diff
@@ -125,7 +119,7 @@ func partitionSeriesByDay(from, through model.Time, seriesWithChunks []*logproto
 			})
 
 			// All chunks fall outside of the range
-			if min == len(chunks) || max == 0 {
+			if min == len(chunks) || max == 0 || min == max {
```

Comment:
idea: I feel we have run already too many times into forgetting to check `min==max` maybe we should wrap this search use-case into some Template function or something.